src/server/config_registry.cc

```diff
@@ -0,0 +1,44 @@
+// Copyright 2023, Roman Gershman.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+#include "src/server/config_registry.h"
+
+#include <absl/flags/reflection.h>
+
+#include "base/logging.h"
+
+namespace dfly {
+
+using namespace std;
+
+ConfigRegistry& ConfigRegistry::Register(std::string_view name, WriteCb cb) {
+  absl::CommandLineFlag* flag = absl::FindCommandLineFlag(name);
+  CHECK(flag) << "Unknown config name: " << name;
+
+  unique_lock lk(mu_);
+  auto [it, inserted] = registry_.emplace(name, std::move(cb));
+  CHECK(inserted) << "Duplicate config name: " << name;
+  return *this;
+}
+
+// Returns true if the value was updated.
+bool ConfigRegistry::Set(std::string_view config_name, std::string_view value) {
```

Comment:
An idea, potentially for the future:
The limitation of the registry as it is present, is that it is tightly integrated with flags. I doesn't support other types of config (i.e. not exposed via flag).
We could add (again, maybe in a future commit) a handler for setting the string, with a default function that sets the flag, but that is override-able by the registry.

Reply:
i agree with both comments